docs/proposals/odo-debug.md

```diff
@@ -0,0 +1,82 @@
+# odo debug support proposal
+
+Table of Contents:
+- [odo debug support proposal](#odo-debug-support-proposal)
+  - [Design overview](#design-overview)
+    - [New odo commands](#new-odo-commands)
+      - [`odo debug start`](#odo-debug-start)
+      - [`odo debug stop`](#odo-debug-stop)
+      - [`odo debug status`](#odo-debug-status)
+    - [`odo debug port-forward LOCAL_PORT:REMOTE_PORT`](#odo-debug-port-forward-localportremoteport)
+- [Languages support](#languages-support)
+  - [NodeJS](#nodejs)
+    - [Problem 1](#problem-1)
+      - [Solution 1](#solution-1)
+      - [Solution 2](#solution-2)
+  - [Java](#java)
+
+## Design overview
+
+The debug mode inside the container is controlled by `DEBUG_MODE` and `DEBUG_PORT` env variable.
+- if `DEBUG_MODE == true` than the code should be executed with remote debugging capabilities enabled.
+- `DEBUG_PORT` is optional and controls the port of the remote debugger.
```

Comment:
nope, the application will still start on port 8080, or how it is configured.

`DEBUG_PORT` controls additional port that will be opened for the debugger connections


for nodejs the run command will look something like this:
```
nodejs --inspect ${DEBUG_PORT} server.js
```
for java
```
java -Xdebug -Xrunjdwp:transport=dt_socket,address=${DEBUG_PORT},server=y,suspend=y
```

Reply:
OK so besides running the app on 8080, port 9090 in the container will also be exposed and listening for some kind of connections. And the debug client (IDE in case of Adapters) will connect to this 9090 port. Is that understanding correct?

Reply:
Yes, that is correct.